File: Hivemind/Entities/GangerWeapon.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Hivemind.Entities
{
    public class GangerWeapon
    {
        public string GangerWeaponId { get; set; }
        public string GangerId { get; set; }
        public Weapon Weapon { get; set; }

        /// <summary>
        /// Cost is set here as an integer, and represents the amount paid for the item.
        /// Some weapons have variable cost, and for purposes of calculating gang rating 
        /// we need the exact number.
        /// </summary>
        public int Cost { get; set; }

        public GangerWeapon()
        {
            Weapon = new Weapon();
        }
    }
}

File: Hivemind.Ng/src/autogenerated/entities/GangerWeapon.ts
import { Weapon } from './Weapon';

export class GangerWeapon {
    public gangerWeaponId: string;
    public gangerId: string;
    public weapon: Weapon;

    public constructor(partial: Partial<GangerWeapon>) {
        if (partial.gangerWeaponId) {
            this.gangerWeaponId = partial.gangerWeaponId;
        }
        if (partial.gangerId) {
            this.gangerId = partial.gangerId;
        }
        if (partial.weapon) {
            this.weapon = partial.weapon;
        }
    }
}